flutter/test/span_frame_metrics_collector_test.dart

```diff
@@ -0,0 +1,242 @@
+    fixture.options.tracesSampleRate = 1.0;
+    fixture.options.addPerformanceCollector(sut);
+    const displayRefreshRate = 60;
+
+    final tracer = MockSentryTracer();
+
+    final startTimestamp = DateTime.now();
+    when(tracer.startTimestamp).thenReturn(startTimestamp);
+    when(tracer.context).thenReturn(SentryTransactionContext('name', 'op'));
+
+    sut.frames[startTimestamp.add(Duration(milliseconds: 1))] = 500;
+
+    final frameMetrics = sut.computeFrameMetrics(tracer,
+        startTimestamp.add(Duration(milliseconds: 10)), displayRefreshRate);
+
+    expect(frameMetrics.isEmpty, isTrue);
+  });
+
+  test('calculates frame metrics correctly for multiple simultaneous spans',
+      () async {
+    final sut = fixture.sut;
+    fixture.options.tracesSampleRate = 1.0;
+    fixture.options.addPerformanceCollector(sut);
+
+    final tracer1 =
+        SentryTracer(SentryTransactionContext('name1', 'op1'), fixture.hub);
+    final tracer2 =
+        SentryTracer(SentryTransactionContext('name2', 'op2'), fixture.hub);
+
+    await Future<void>.delayed(Duration(milliseconds: 800));
+    await tracer1.finish();
+
+    await Future<void>.delayed(Duration(milliseconds: 800));
+    await tracer2.finish();
+
+    expect(tracer1.data['frames.slow'], 2);
+    expect(tracer1.data['frames.frozen'], 1);
+    expect(
+        tracer1.data['frames.delay'], _isWithinRange(expectedFramesDelay, 10));
+    expect(
+        tracer1.data['frames.total'], _isWithinRange(expectedTotalFrames, 2));
+
+    expect(tracer1.measurements['frames_delay']!.value,
+        _isWithinRange(expectedFramesDelay, 10));
+    expect(tracer1.measurements['frames_total']!.value,
+        _isWithinRange(expectedTotalFrames, 2));
+    expect(tracer1.measurements['frames_slow']!.value, 2);
+    expect(tracer1.measurements['frames_frozen']!.value, 1);
+
+    expect(tracer2.data['frames.slow'], 2);
+    expect(tracer2.data['frames.frozen'], 1);
+    expect(
+        tracer2.data['frames.delay'], _isWithinRange(expectedFramesDelay, 10));
+    // expect(tracer2.data['frames.total'], _isWithinRange(expectedTotalFrames, 2));
+
+    expect(tracer2.measurements['frames_delay']!.value,
+        _isWithinRange(expectedFramesDelay, 10));
+    // expect(tracer2.measurements['frames_total']!.value,
+    //     _isWithinRange(expectedTotalFrames, 2));
+    expect(tracer2.measurements['frames_slow']!.value, 2);
+    expect(tracer2.measurements['frames_frozen']!.value, 1);
+  });
+
+  test('frame tracker is paused after finishing a span', () async {
+    final sut = fixture.sut;
+    fixture.options.tracesSampleRate = 1.0;
+    fixture.options.addPerformanceCollector(sut);
+
+    final tracer =
+        SentryTracer(SentryTransactionContext('name', 'op'), fixture.hub);
+
+    await Future<void>.delayed(Duration(milliseconds: 800));
+
+    await tracer.finish();
+
+    expect(sut.isFrameTrackingPaused, isTrue);
+  });
+}
+
+Matcher _isWithinRange(int expected, [int delta = 5]) => allOf(
```

Comment:
can you recommend any sensible ranges? machines running these tests will have different performances so it might be flaky if we don't add good ranges